webpack/assets/javascripts/react_app/routes/common/reducerHOC/withQueryReducer.js

```diff
@@ -0,0 +1,23 @@
+import Immutable from 'seamless-immutable';
+
+const initialState = Immutable({
+  page: 1,
+  perPage: 25,
```

Comment:
there is no way to access server props from here, maybe we should agree on a default constant. per page options are in redux though.

Reply:
Perhaps we need to find a way? I doubt this is the last place where we care about some setting or attribute from the server in the initial state.

Reply:
The option I can think of is to remove `perPage` here, when a page mounts for the first time it will fetch the perPage from redux and merge it into its state.